Watchman.Discord/Areas/Help/Services/HelpService.cs

```diff
@@ -1,34 +1,76 @@
+        private readonly HelpInformationFactory _helpInformationFactory;
         private readonly ISession _session;
 
-        public HelpService(ISessionFactory sessionFactory)
+        public HelpService(ISessionFactory sessionFactory, IQueryBus queryBus, HelpInformationFactory helpInformationFactory)
         {
+            _queryBus = queryBus;
+            _helpInformationFactory = helpInformationFactory;
             _session = sessionFactory.Create();
         }
 
+        public string GenerateHelp(Contexts contexts)
+        {
+            var result = this._queryBus.Execute(new GetHelpInformationQuery(contexts.Server.Id));
+
+            var lines = new List<string>();
+            foreach (var helpInfo in result.HelpInformations)
+            {
+                var line = new StringBuilder("-" + helpInfo.Names.Aggregate((x, y) => $"{x} / -{y}"));
+
+                line.Append(" => ");
+                line.Append(helpInfo.Descriptions.First(x => x.Name == helpInfo.DefaultDescriptionName).Details);
+                lines.Add(line.ToString());
+            }
+
+            var messageBuilder = new StringBuilder().PrintManyLines(lines.ToArray());
+            return messageBuilder.ToString();
+        }
+
+        public string GenerateJsonHelp(Contexts contexts)
+        {
+            var result = this._queryBus.Execute(new GetHelpInformationQuery(contexts.Server.Id));
+
+            var serialized = JsonConvert.SerializeObject(result.HelpInformations, Formatting.Indented);
+
+            var messageBuilder = new StringBuilder();
+            messageBuilder.AppendLine("```json");
+            messageBuilder.AppendLine(serialized);
+            messageBuilder.AppendLine("```");
+
+            return messageBuilder.ToString();
+        }
+
         public void FillDatabaseWithNewMethods(IEnumerable<CommandInfo> commandInfosFromAssembly)
         {
+            var commandInfosFromAssemblyList = commandInfosFromAssembly.ToList(); // for not multiple enumerating
             var helpInfosInDb = _session.Get<HelpInformation>().ToList();
 
-            var newCommands = FindNewCommands(commandInfosFromAssembly, helpInfosInDb).ToList();
-            var oldCommands = FindOldHelps(commandInfosFromAssembly, helpInfosInDb).ToList();
-            var changedCommands = FindChangedHelps(commandInfosFromAssembly, helpInfosInDb).ToList();
+            var newCommands = FindNewCommands(commandInfosFromAssemblyList, helpInfosInDb).ToList();
+            var oldCommands = FindOldHelps(commandInfosFromAssemblyList, helpInfosInDb).ToList();
+            var changedCommands = FindChangedHelps(commandInfosFromAssemblyList, helpInfosInDb).ToList();
 
-            var helpInfoFactory = new HelpInformationFactory();
-            newCommands.ForEach(x => _session.Add(helpInfoFactory.Create(x)));
+            newCommands.ForEach(x => _session.Add(this._helpInformationFactory.Create(x)));
             oldCommands.ForEach(x => _session.Delete(x));
```

Comment:
`Fill` nie powinno mieć skutków ubocznych takich jak `Delete`
lepiej ogarniać to dwiema osobnymi metodami, które uruchamia kontroler osobno